testsupport/tiers/nstemplateset.go

```diff
@@ -65,23 +66,18 @@ func UntilNSTemplateSetHasTemplateRefs(expectedRevisions TemplateRefs) wait.NSTe
 	return wait.NSTemplateSetWaitCriterion{
 		Match: func(actual *toolchainv1alpha1.NSTemplateSet) bool {
 			actualNamespaces := actual.Spec.Namespaces
-			if len(actualNamespaces) != len(expectedRevisions.Namespaces) ||
-				expectedRevisions.ClusterResources == nil ||
+			if expectedRevisions.ClusterResources == nil ||
```

Comment:
Wouldn't the length still need to match? So wouldn't the check for `len(actualNamespaces) != len(expectedRevisions.Namespaces)` still be valid as a quick check?

Reply:
I believe that check on the length of the namespaces is implied in the `reflect.DeepEqual(actualNamespaceTmplRefs, expectedRevisions.Namespaces)` call a few lines below

Reply:
yeah, it is but I meant it would return immediately, without having to sort the strings, etc. Just slightly more efficient to do that check up front.

Reply:
right, but we're talking about a slice of 1 or 2 elements, so the benefit of making this check upfront is minimal IMO. Also, this will have to be expended with checks on the space roles in the future, so I believe that checking each section (cluster resource, namespaces and space roles) individually will be simpler to maintain.
But if it's really a concern, I can restore it.

Reply:
Good point, didn't think about space roles too. Agree it's simpler to keep it as is. 👍